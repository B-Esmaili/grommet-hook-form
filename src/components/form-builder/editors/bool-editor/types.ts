import { FormEditorPropsBase } from '../../types';

export type BoolEditorProps = FormEditorPropsBase & {
    defaultValue? : boolean,
    controlType : "checkbox" | "switch"
}