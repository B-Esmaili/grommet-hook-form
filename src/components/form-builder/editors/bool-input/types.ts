import { FormEditorPropsBase } from '../../types';

export type BoolInputProps = FormEditorPropsBase & {
    defaultValue? : boolean,
    controlType : "checkbox" | "switch"
}