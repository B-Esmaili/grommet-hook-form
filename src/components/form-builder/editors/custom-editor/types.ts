import { FormEditorPropsBase } from "../../../form-builder/types";
import React from "react";

export type CustomEditorProps = FormEditorPropsBase & {
  editorComponent: React.ReactElement;
  props?:any
};
