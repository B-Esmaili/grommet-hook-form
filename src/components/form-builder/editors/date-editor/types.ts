import { FormEditorPropsBase } from "../../types";
import {DateInputProps} from "grommet";

export type DateEditorProps = FormEditorPropsBase & {
    minDate?: string,
    maxDate?: string,
    dateInputProps?:DateInputProps
    defaultValue?: string | Date
}