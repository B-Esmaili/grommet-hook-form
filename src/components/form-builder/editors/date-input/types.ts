import { FormEditorPropsBase } from "../../types";

export type DateInputProps = FormEditorPropsBase & {
    minDate?: string,
    maxDate?: string,
    defaultValue: string | number
}