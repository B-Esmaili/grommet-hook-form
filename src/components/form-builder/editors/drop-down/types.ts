import { AxiosRequestConfig } from "axios";
import MockAdapter from "axios-mock-adapter/types";
import React from "react";
import { SelectProps } from "grommet";

export interface RemoteDataSource {
  request: string | AxiosRequestConfig;
  searchKey?: string;
  pageKey?: string;
  pageSizeKey?: string;
  pageSize?: number;
  extraParams?: any;
  onResponse?: (data: any, headers: any) => any;
  onRequest?: (data: any, header: any) => any;
  listKey?: string;
  totalKey?: string;
  mockResponse?: (req: MockAdapter) => void;
}

export interface OptionProps {
  label: React.ReactNode;
  selected: boolean;
}

export interface DropDownOperations {
  setValue: (setter: (value: any[] | any) => any[] | any) => void;
}

export interface DropDownProps {
  itemValueKey: string;
  itemLabelKey: string;
  labelWrap?: React.ReactElement;
  plainLabel?: boolean;
  renderItemLabel?: <TOption = any>(
    option: TOption,
    operations: DropDownOperations,
    index: number
  ) => React.ReactNode;
  multiple?: boolean;
  options: any[] | RemoteDataSource;
  placeholder?: React.ReactNode;
  onSearch?: (text: string, options: any[]) => any[];
  renderItem?: <TOption = any>(options: TOption,selected : boolean) => React.ReactNode;
  selectProps?: SelectProps;
  searchDebounce?: number;
}
