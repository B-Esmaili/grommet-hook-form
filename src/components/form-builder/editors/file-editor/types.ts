import { FileInputProps  } from "grommet";

export interface FileEditorProps {
  multiple?: boolean;
  onRemove?:(files : File[])=>void,
  fileInputProps?: Omit<
  FileInputProps,
    "onChange" | "messages" | "multiple" | "ref" | "name"
  >;
}
