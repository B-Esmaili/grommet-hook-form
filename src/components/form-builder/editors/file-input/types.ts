import { FileInputProps as GrommetFileInputProps } from "grommet";

export interface FileInputProps {
  multiple?: boolean;
  fileInputProps?: Omit<
    GrommetFileInputProps,
    "onChange" | "messages" | "multiple" | "ref" | "name"
  >;
}
