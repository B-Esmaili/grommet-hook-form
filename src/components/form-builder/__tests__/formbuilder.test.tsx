import { fireEvent, getByTestId, getByText, render } from "@testing-library/react";
import "jest-styled-components";
import { act } from "react-dom/test-utils";
import { FormBuilder } from "../form-builder";
import { FormFieldType } from "../types";

describe("FormBuilder", () => {

  it("model overrides default value", () => {
    const DropDownField = () => {
      return (
        <div>
          <FormBuilder
            model={{
              test: "value2",
            }}
            fields={[
              {
                type: FormFieldType.Text,
                defaultValue: "value1",
                name: "test",
                label: "test",
              },
            ]}
          >
            {(methods) => (
              <div data-testid="value">{methods.watch("test")}</div>
            )}
          </FormBuilder>
        </div>
      );
    };

    let { container } = render(<DropDownField />);

    let value = getByTestId(container, "value").innerHTML;
    expect(value).toBe("value2");
  });

  it("onChange event fires when value changes", () => {
    jest.useFakeTimers();
    window.HTMLElement.prototype.scrollIntoView = jest.fn();
    window.scrollTo = jest.fn();

    const Form = ({ onChange }: any) => {
      return (
        <div>
          <FormBuilder
            fields={[
              {
                type: FormFieldType.DropDown,
                options: [
                  { text: "Opt1", value: "opt1" },
                  { text: "Opt2", value: "opt2" },
                ],
                itemLabelKey: "text",
                itemValueKey: "value",
                name: "test",
                label: "test",
                placeholder: "dropdown",
                onChange: (v) => onChange && onChange(v),
              },
            ]}
          />
        </div>
      );
    };

    let onChangeFn = jest.fn();

    const { container } = render(<Form onChange={onChangeFn} />);

    let dp = getByText(container, "dropdown");
    fireEvent.click(dp);

    act(() => {
      jest.advanceTimersByTime(200);
    });

    let secondOption = getByText(document.body, "Opt2");

    fireEvent.click(secondOption);

    act(() => {
      jest.advanceTimersByTime(100);
    });

    expect(onChangeFn).toBeCalledWith("opt2");
  });
});

export {};
