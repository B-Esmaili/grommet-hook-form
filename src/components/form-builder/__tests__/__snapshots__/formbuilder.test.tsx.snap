// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBuilder renders default value 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: body;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  grid-area: no-area;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c0 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  grid-template-areas: "body" "actions";
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 96px;
}

.c8 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c8:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c8:focus > circle,
.c8:focus > ellipse,
.c8:focus > line,
.c8:focus > path,
.c8:focus > polygon,
.c8:focus > polyline,
.c8:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c8:focus::-moz-focus-inner {
  border: 0;
}

.c8::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c8::-moz-placeholder {
  color: #AAAAAA;
}

.c8:-ms-input-placeholder {
  color: #AAAAAA;
}

.c8::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c8::-moz-focus-inner {
  border: none;
  outline: none;
}

.c8:-moz-placeholder,
.c8::-moz-placeholder {
  opacity: 1;
}

.c7 {
  position: relative;
  width: 100%;
}

.c6 {
  width: 15em;
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    margin-top: 3px;
    margin-bottom: 3px;
  }
}

<div
  class=""
>
  <form
    fields="[object Object]"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4 "
            >
              <label
                class="c5"
              >
                <label
                  class="c6"
                >
                  test
                </label>
              </label>
              <div
                class="c3 FormField__FormFieldContentBox-m9hood-1"
              >
                <div
                  class="c7"
                >
                  <input
                    autocomplete="off"
                    class="c8"
                    value="value1"
                  />
                </div>
              </div>
              <div
                class="c9"
                color="status-critical"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
